safekeeper/src/send_interpreted_wal.rs

```diff
@@ -94,9 +94,14 @@ impl<IO: AsyncRead + AsyncWrite + Unpin> InterpretedWalSender<'_, IO> {
                         }
                     }
 
+                    let max_next_record_lsn = match max_next_record_lsn {
```

Comment:
> Or do we want to send an empty batch to signal a new max_next_record_lsn in this case?

Exactly. Records may be empty because they're not intended for _this_ shard, but we need to send _something_
to tell the pageserver to advance its last record LSN.